src/core/server/status/types.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Observable } from 'rxjs';
+
+/**
+ * The current status of a service at a point in time.
+ *
+ * @typeParam Meta - JSON-serializable object. Plugins should export this type to allow other plugins to read the `meta`
+ *                   field in a type-safe way.
+ * @public
+ */
+export type ServiceStatus<Meta extends Record<string, any> | unknown = unknown> =
+  | {
+      /**
+       * The current availability level of the service.
+       */
+      level: ServiceStatusLevel.available;
+      /**
+       * A high-level summary of the service status.
+       */
+      summary?: string;
+      /**
+       * A more detailed description of the service status.
+       */
+      detail?: string;
+      /**
+       * A URL to open in a new tab about how to resolve or troubleshoot the problem.
+       */
+      documentationUrl?: string;
+      /**
+       * Any JSON-serializable data to be included in the HTTP API response. Useful for providing more fine-grained,
+       * machine-readable information about the service status. May include status information for underlying features.
+       */
+      meta?: Meta;
+    }
+  | {
+      level: ServiceStatusLevel;
+      summary: string; // required when level !== available
+      detail?: string;
+      documentationUrl?: string;
+      meta?: Meta;
```

Comment:
NIT: these three last properties are the same in both cases. They could be factorized, even if I'm not sure this would increase readability on this already complex type.

Reply:
Resolved by reducing to a single interface with `level` and `summary` always required.